dhall/src/Dhall/Diff.hs

```diff
@@ -745,7 +754,22 @@ diffAnnotatedExpression l r =
     diffOperatorExpression l r
 
 diffOperatorExpression :: (Eq a, Pretty a) => Expr Void a -> Expr Void a -> Diff
-diffOperatorExpression = diffOrExpression
+diffOperatorExpression = diffImportAltExpression
+
+diffImportAltExpression :: (Pretty a, Eq a) => Expr Void a -> Expr Void a -> Diff
+diffImportAltExpression l@(ImportAlt {}) r@(ImportAlt {}) =
+    enclosed' "    " (operator "?" <> "  ") (docs l r)
```

Comment:
Am I using the right amount of space here? It seems to vary between the various `diff*Expression`s but I'm not sure why.

Reply:
All indentation (whether pretty-printing or diffing) is a multiple of two spaces, so if the operator is one character long (like `?`) then the diff pads the left margin to two space:

```
␣␣e₀
?␣e₁
```

... but if the operator is two characters long (like `||`) then the diff pads the left margin to four spaces:

```
␣␣␣␣e₀
||␣␣e₁
```

So in this case what you want is:

```haskell
enclosed' "  " (operator "?" <> " ") (docs l r)
```

Reply:
Thanks! I've added your explanation as a comment on `diffOperatorExpression.